packages/mux-active-viewer-count/src/index.ts

```diff
@@ -0,0 +1,225 @@
+            return Promise.reject(respObj?.error ?? 'no data in response');
+          }
+          // Otherwise, we successfully retrieved the latest views, so
+          // provide that info out via `callback()`.
+          callback(views);
+          return views;
+        })
+        // Catch and invoke errorCb before timeout + re-fetch. This allows
+        // for re-fetching by default, but provides the opportunity
+        // to unsubscribe externally via `errorCb` if desired (CJP)
+        .catch(errorCb)
+        // Wait the duration of the polling interval before restarting
+        // the next fetch
+        .then(() => {
+          return new Promise((resolve) => {
+            timeoutId = setTimeout(async () => {
+              resolve(undefined);
+            }, pollInterval * 1000);
+          });
+        })
+        // Restart process of re-fetching by invoking this method again
+        // (async recursion)
+        .then(fetchViewerCountPoll)
+    );
+  };
+
+  // Kick off the polling functionality.
+  fetchViewerCountPoll();
+
+  // Return an "unsubscribe()" function. Invoking this will abort
+  // any mid-flight fetches, clear any pending timeouts for
+  // a re-fetch, and mark this process as "aborted".
+  return () => {
+    aborted = true;
+    controller.abort();
+    if (typeof timeoutId === 'number') {
+      clearTimeout(timeoutId);
+      timeoutId = undefined;
+    }
+  };
+};
+
+const Attributes = {
+  TOKEN: 'token',
+  POLL_INTERVAL: 'poll-interval',
+  DISABLED: 'disabled',
+};
+
+const DEFAULT_POLL_INTERVAL = 20;
+
+const AttributeValues = Object.freeze(Object.values(Attributes));
+
+const ARIA_LABEL = 'Viewer Count';
+const DEFAULT_VIEWER_COUNT_VALUE = '??';
+const toAriaLabel = ({ viewerCount }: { viewerCount?: number }) => {
+  const viewerCountStr = typeof viewerCount === 'number' ? `${viewerCount}` : 'Unknown';
+  return `${ARIA_LABEL} ${viewerCountStr}`;
+};
+
+const template = document.createElement('template');
+template.innerHTML = `
+<style>
+</style>
+<span id="viewer-count">
+${DEFAULT_VIEWER_COUNT_VALUE}
+</span>
+`;
+
+class MuxActiveViewerCountElement extends globalThis.HTMLElement {
+  static get observedAttributes() {
+    return AttributeValues;
+  }
+
+  #unsubscribeViewerCount: (() => void) | undefined;
+
+  get #viewerCountEl() {
+    return (this.shadowRoot as ShadowRoot).querySelector('#viewer-count') as HTMLElement;
+  }
+
+  #views?: number
+
+  get views(): number {
+    return this.#views ?? Number.NaN;
+  }
+
+  constructor() {
+    super();
+    const shadowRoot = this.attachShadow({ mode: 'open' });
+    shadowRoot.appendChild(template.content.cloneNode(true));
+  }
+
+  get token() {
+    return this.getAttribute(Attributes.TOKEN) ?? '';
+  }
+
+  set token(value: string) {
+    if (this.token === value) return;
+    this.#views = undefined;
+    this.setAttribute(Attributes.TOKEN, value);
+  }
+
+  get pollInterval() {
+    let temp = this.hasAttribute(Attributes.POLL_INTERVAL)
```

Comment:
suggestion(fnon-blocking): We may want to just add a private `#pollInterval` and handle this via the setter instead. Can be a followup.